roles/update_etc_hosts/tasks/update_host_present.yml

```diff
@@ -0,0 +1,149 @@
+
+# We allow more than one line containing sap_ip:
+- name: Check for duplicate entries of {{ thishost.node_ip }}
+  ansible.builtin.shell: |
+    n=$(grep "^{{ thishost.node_ip }}\s" {{ __update_etc_hosts_file }} | wc -l)
+    if [ $n -gt 1 ]; then
+      echo "Duplicate IP entry in {{ __update_etc_hosts_file }}!"
+      exit 1
+    else
+      exit 0
+    fi
+  register: __update_etc_hosts_register_duplicate_ip_check
+  changed_when: false
+  ignore_errors: true
+  when: not ansible_check_mode
+
+- name: Report if there is more than one line with the IP address
+  ansible.builtin.debug:
+    msg:
+      - "More than one line containing {{ thishost.node_ip }}. File {{ __update_etc_hosts_file }} will not be modified."
+  when:
+    - not ansible_check_mode
+    - __update_etc_hosts_register_duplicate_ip_check.stdout == 'Duplicate IP entry in {{ __update_etc_hosts_file }}!'
+
+- name: Ensure that the entry in hosts file is correct
+  ansible.builtin.lineinfile:
+    path: "{{ __update_etc_hosts_file }}"
+    regexp: '^{{ thishost.node_ip }}\s'
+    line: "{{ thishost.node_ip }} {{ thishost.node_name }}.{{ __update_etc_hosts_domain }} {{ thishost.node_name }} {{ __update_etc_hosts_aliases }} {{ __update_etc_hosts_comment }}"
+    backup: true
+  when:
+    - not ansible_check_mode
+    - __update_etc_hosts_register_duplicate_ip_check.stdout != 'Duplicate IP entry in {{ __update_etc_hosts_file }}!'
+
+- name: Check for duplicate or missing entries of hostname and fqdn in {{ __update_etc_hosts_file }}
+  ansible.builtin.shell: |
+    n=$(awk 'BEGIN{a=0}/^{{ line_item }}\s/||/\s{{ line_item }}\s/||/\s{{ line_item }}$/{a++}END{print a}' {{ __update_etc_hosts_file }})
+    if [ $n -eq 1 ]; then
+      exit 0
+    else
+      exit 1
+    fi
+  loop:
+    - '{{ thishost.node_name }}.{{ __update_etc_hosts_domain }}'
+    - '{{ thishost.node_name }}'
+  changed_when: false
+  loop_control:
+    loop_var: line_item
+  when: not ansible_check_mode
+
+- name: Check if ipv4 address, FQDN, and hostname are in {{ __update_etc_hosts_file }}
+  block:
+    - name: Perform the hosts file completeness check
+      ansible.builtin.command: awk 'BEGIN{a=0}/{{ thishost.node_ip }}/&&/{{ thishost.node_name }}.{{ __update_etc_hosts_domain }}/&&/{{ thishost.node_name }}/{a++}END{print a}' {{ __update_etc_hosts_file }}
+      register: __update_etc_hosts_register_ipv4_fqdn_sap_hostname_once_check
+      changed_when: false
+
+    - name: Display the output of the hosts file completeness check
+      ansible.builtin.debug:
+        var: __update_etc_hosts_register_ipv4_fqdn_sap_hostname_once_check.stdout_lines,
+              __update_etc_hosts_register_ipv4_fqdn_sap_hostname_once_check.stderr_lines
+
+    - name: Display the expected output of the hosts file completeness check
+      ansible.builtin.debug:
+        msg:
+          - "Expected:"
+          - "{{ thishost.node_ip }} {{ thishost.node_name }}.{{ __update_etc_hosts_domain }} {{ thishost.node_name }}"
```

Comment:
why comments and aliases are not expected?

Reply:
It is the minimum expected. Will change

Reply:
corrected error message